csrc/scheduler/greedy.cpp

```diff
@@ -319,6 +321,28 @@ class CompileTimeChecker : private IterVisitor {
     checkConstrainedTv(ir_utils::getTvOutput(pad), pad->getPaddedAxes());
   }
 
+  void handle(TopKOp* topk) override {
+    // Due to the current limitations of TopKOp codegen, all of the
+    // TIDx threads participate, which means the TID parallelized iter
+    // domain of this TopKOp must have an extent that is no less
+    // than any other TID parallelized iter domains. Requiring the
```

Comment:
> which means the TID parallelized iter domain of this TopKOp must have an extent that is no less than any other TID parallelized iter domains.

For my own education, if we have TID extent on top k being smaller, we ended up with garbage data participating in sorting, so predicating on the consumer of top k (aka sort) output would produce wrong results.